New and improved Examples collection

escherize opened the pull request:
We want to ship a new and improved Examples collection. That gets loaded during startup, in a custom migration. We read from a bespoke edn format and load it into the app. We only run this when there is nothing else in the DB, so we can make some simplifying assumptions (e.g. hardcoding ids etc, which serialization handles in a different way.) This edn file has the information needed to populate the new and improved Examples collection contents into the app-db during startup migrations by the `CreateSampleContent` migration.

# How to see the content:

Start a fresh metabase install, and you should see the Examples collection:

<img width="1432" alt="Screenshot 2024-12-03 at 4 46 05 PM" src="https://github.com/user-attachments/assets/16377543-8709-43f3-9e79-13df3d31cf76">

escherize committed:
fix the script, and add the new edn

escherize committed:
Merge branch 'master' into update-sample-collection-migration-load

escherize committed:
make the migrations run in the proper order

escherize committed:
it works (needs cleanup)

escherize committed:
Merge branch 'update-sample-collection-migration-load' of github.com:metabase/metabase into update-sample-collection-migration-load

escherize committed:
remove broken binning

- These bins rely on fingerprinting for the custom Age column

- The fingerprinting for that custom column are not properly exported in
serialization, hence do not get loaded into metabase, causing binning by
Age to fail.

escherize committed:
cleanup / linting

escherize committed:
cleanup

escherize committed:
Merge branch 'master' into update-sample-collection-migration-load

escherize committed:
update migration test ids to test

cdeweyx commented:
Looking real nice 🙏

Two blocking things I found:
- Examples should be the top level collection. You can kill the parent "Product" collection, that's cruft from where the content was on stats. Main blocker in my eyes.

<img width="415" alt="Screenshot 2024-12-04 at 17 11 01" src="https://github.com/user-attachments/assets/13e879f6-ebd0-4f17-9112-3cbfa510a596">

* Click behavior for the "Revenue goal for this quarter" progress chart ended up breaking somehow. It should link to the "Portfolio Performance" dashboard tab, right now it's empty. If this is complex in the interest of shipping, we can remove the click behavior but would be good to have.

<img width="901" alt="Screenshot 2024-12-04 at 17 12 04" src="https://github.com/user-attachments/assets/9b11b0a9-e290-45f1-a351-e458c239d4d8">

escherize committed:
realign test name to check for to see if sample content was created

cdeweyx reviewed (changes requested):
Looking real nice 🙏

Two blocking things I found:
- Examples should be the top level collection. You can kill the parent "Product" collection, that's cruft from where the content was on stats. Main blocker in my eyes.

<img width="415" alt="Screenshot 2024-12-04 at 17 11 01" src="https://github.com/user-attachments/assets/13e879f6-ebd0-4f17-9112-3cbfa510a596">

* Click behavior for the "Revenue goal for this quarter" progress chart ended up breaking somehow. It should link to the "Portfolio Performance" dashboard tab, right now it's empty. If this is complex in the interest of shipping, we can remove the click behavior but would be good to have.

<img width="901" alt="Screenshot 2024-12-04 at 17 12 04" src="https://github.com/user-attachments/assets/9b11b0a9-e290-45f1-a351-e458c239d4d8">

escherize committed:
remove Product collection + move Examples into root

escherize committed:
update collection_ids for the content too

escherize committed:
remove click behavior for 'Revenue goal for this quarter'

escherize commented:
Thanks @cdeweyx.

I've:

1. moved the Example collection into the root, reparented its content to match the new collection id, and removed the Products collection.

2. I tried fixing the click-thru behavior, but something about the stringified viz settings wasn't obvious. So I dropped the click-thru for ['Revenue goal for this quarter'](https://github.com/metabase/metabase/pull/50829/commits/f53a9e3d42e28662af645e3ab317f035b44f2d6c).

crisptrutski commented on src/metabase/db/custom_migrations.clj:
```diff
@@ -1157,6 +1157,10 @@
 (defn- internal-user-exists? []
   (pos? (first (vals (t2/query-one {:select [:%count.*] :from :core_user :where [:= :id config/internal-mb-user-id]})))))
 
+(define-migration CreateInternalUserNoOp
```
I couldn't see where this migration is triggered, do you still need it?

escherize replied:
good call, removed.

crisptrutski commented on test/metabase/db/custom_migrations_test.clj:
```diff
@@ -1981,55 +1981,57 @@
+          (binding [custom-migrations/*create-sample-content* create?]
+            (is (false? (sample-content-created?)))
+            (migrate!)
+            (if create?
+              (is (true? (sample-content-created?)))
+              (is (false? (sample-content-created?)))))
```
Have to have at least one subjective nit

```suggestion
            (is (= create? sample-content-created?))
```

escherize replied:
Haha, thanks. My goal was to make the test print out what failed, instead of a report about:

`((if create? true? false?) (sample-content-created?)) failed` 

I'll take your suggestion since I think it does that too.

crisptrutski reviewed (approved):
LGTM from the migration code side

escherize committed:
Apply suggestions from code review

Co-authored-by: Chris Truter <crisptrutski@users.noreply.github.com>

escherize committed:
for the sample collection is_sample should be true

escherize committed:
Drop CreateInternalUserNoOp

escherize committed:
Merge branch 'update-sample-collection-migration-load' of github.com:metabase/metabase into update-sample-collection-migration-load

escherize committed:
typo

escherize committed:
update migration comments

escherize committed:
2 small fixes

escherize committed:
fix data perms for example coll

escherize committed:
add a comment so we don't have 2 blank lines in a row
